src/Compiler/pars.fsy

```diff
@@ -2762,6 +2762,10 @@ unionCaseReprElement:
      { let xmlDoc = grabXmlDoc(parseState, [], 1)
        mkSynAnonField ($1, xmlDoc) }
 
+  | invalidUseOfAppTypeFunction
```

Comment:
Could you please also add a production with `ident COLON`, so `| A of x: int -> int` is also parsed? See the first production above.